### PR TITLE
Add doc about GroupData, TopicData.. on XML [14087]

### PR DESCRIPTION
**Pull request by richiware**
Depends on:

* eProsima/Fast-DDS#2554

### CODE REVIEW

File: code/DDSCodeTester.cpp [hunk 1]
```diff
@@ -3197,8 +3197,11 @@ void dds_qos_examples()
         std::vector<eprosima::fastrtps::rtps::octet> vec;
         vec = group_data.data_vec(); // Getter function
 
+        //Add two new octets to topic data vector
```
Comment (MiguelCompany) [inline code/DDSCodeTester.cpp:3200]:
```suggestion
        //Add two new octets to group data vector
```

File: code/DDSCodeTester.cpp [hunk 2]
```diff
@@ -3325,9 +3330,11 @@ void dds_qos_examples()
         std::vector<eprosima::fastrtps::rtps::octet> vec;
         vec = user_data.data_vec(); // Getter Function
 
-        //Add new octet to topic data vector
+        //Add two new octets to topic data vector
```
Comment (MiguelCompany) [inline code/DDSCodeTester.cpp:3333]:
```suggestion
        //Add two new octets to user data vector
```

File: code/XMLTester.xml [hunk 3]
```diff
@@ -4129,6 +4131,60 @@
...
+    </qos>
+</data_writer>
+
+<data_reader profile_name="reader_xml_conf_userdata_profile">
+    <qos>
+        <userData>
+            <value>3.a</value>
+        </userData>
+    </qos>
+</data_reader>
```
Comment (MiguelCompany) [inline code/XMLTester.xml:4185]: User data is also a participant QoS

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

File: code/XMLTester.xml (hunk 3 above)
Comment (richiware) [reply code/XMLTester.xml:4193]: Done

**Review by MiguelCompany — APPROVED** [review]
LGTM

**Pull request merged**